### ISSUE TITLE
toToml filter converts integers and floats to quoted-strings

### ISSUE THREAD

**Issue by johanfleury** [helm/helm#6539]
# Bug Report

When using the `toToml` filter in a template, integers and floats are converted to quoted-strings (which is not the case with `toYaml`).

This lead to issues with other TOML decoders (`telegraf` will crash with: `E! [telegraf] Error running agent: Error parsing /etc/telegraf/telegraf.conf, line 41: (config.AgentConfig.MetricBatchSize) cannot unmarshal TOML string into int`).

# Example

The following “Chart”:

```
$ tree
.
├── Chart.yaml
├── templates
│   └── test.yaml
└── values.yaml

1 directory, 3 files

$ cat Chart.yaml
---
name: test
version: 0.1.0

$ cat values.yaml
---
config:
  test1: 1

  test2:
    test2: 1

  test3:
    test3:
      - test3: 1

  test4: 1.1

  test5:
    test5: 1.1

  test6:
    test6:
      - test6: 1.1

$ cat templates/test.yaml
{{ .Values.config | toToml }}
{{ .Values.config | toYaml }}
```

Produces:

```
$ helm template .
---
# Source: test/templates/test.yaml
test1 = "1"
test4 = "1.1"

[test2]
  test2 = "1"

[test3]

  [[test3.test3]]
    test3 = "1"

[test5]
  test5 = "1.1"

[test6]

  [[test6.test6]]
    test6 = "1.1"

test1: 1
test2:
  test2: 1
test3:
  test3:
  - test3: 1
test4: 1.1
test5:
  test5: 1.1
test6:
  test6:
  - test6: 1.1
```

# Related issues / pull requests

This issue was introduced in PR #6010 which was made to fix #1707 and #2474.

# Installation details

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.14+unreleased", GitCommit:"3b51dfbdbcf34c9ece3e6840dad0ab70cc13c1c2", GitTreeState:"clean"}                                                                                    
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}   
```

Output of `kubectl version`: **not relevant**

Cloud Provider/Platform (AKS, GKE, Minikube etc.): **not relevant**

**Comment by bacongobbler** [comment]
@johanfleury [I mentioned](https://github.com/helm/helm/issues/2474#issuecomment-536687114) that https://github.com/helm/helm/pull/6010 likely FIXED this issue. Not introduced it. As requested earlier, can you please check and confirm with the canary release. Thanks.

https://helm.sh/docs/using_helm/#from-canary-builds

**Comment by johanfleury** [comment]
That's what I did, and the difference between 2.14.3 and the canary release (2.14 commit `3b51dfbdbcf34c9ece3e6840dad0ab70cc13c1c2`) is that intergers that where converted to floats are now converted to quoted-string.

**Comment by johanfleury** [comment]
I did some tests, and the issue is due to the use of `json.Number` for floats and integers unmarshaling: https://play.golang.org/p/afaXYox7qBP.

I'm not really into Go so I not 100% sure, but I suppose the [TOML library](https://github.com/BurntSushi/toml) does not support `json.Number` (I'm not sure if it should).

**Comment by johanfleury** [comment]
@bacongobbler would you mind requalifying this issue as bug to give it the visibility it deserves please?

**Comment by bacongobbler** [comment]
fixed via #6749.

**Comment by johanfleury** [comment]
So what's the current situation? Are integers still converted to float as it was before? If that's the case, should we open a new issue or reopen the old ones?
